src/classes/OPP_AutomatedSoftCreditsService.cls

```diff
@@ -0,0 +1,51 @@
+/*
+    Copyright (c) 2018, Salesforce.org
+    All rights reserved.
+    
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+    
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+* @author Salesforce.org
+* @date 2018
+* @group Opportunity
+* @description Handles Automated Soft Credits for Individual and Organizational gifts.
+*/
+
+public class OPP_AutomatedSoftCreditsService {
+
+	/*********************************************************************************************************
+    * @description Determines if the Account is an Organization based on its account classification, 
+    *				e.g., Individual.
+    * @param accountClassification The classification of an Opportunity's associated Account.
+    * @return Boolean
+    **********************************************************************************************************/
+    public static Boolean isOrganizationalAccount(Boolean accountClassification) {
```

Comment:
I see where you're going. However, since the code is being called internally, as opposed to from an outside method, l I think the system field is enough rather than pass the entire Account object when only that attribute is necessary. Also, I think this method is better than looking directly at the system field because it improves readability. When someone is looking at the code, they can read isOrganizationalAccount and understand what's happening quicker than if they just saw the system field.

Reply:
I agree your creating the isOrgAccount() is a good thing!  i hate that we have a bunch of code looking at these system flags (that I always forget which ones to look at).  I just want the abstraction to be real, rather than forcing the callers to still pass the correct system flag.

Reply:
Yes, I get that. Let's put some cycles on that one.